### PR TITLE
Plenigo 1005

### PR DESCRIPTION
**Pull request by rtorres87**

### CODE REVIEW

File: src/main/java/com/plenigo/sdk/models/CompanyUser.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
+package com.plenigo.sdk.models;
+
+/**
+ * Created by rtorres on 03/02/16.
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/models/CompanyUser.java]: Comment

File: src/main/java/com/plenigo/sdk/models/CompanyUserBillingAddress.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.plenigo.sdk.models;
+
+/**
+ * Created by rtorres on 03/02/16.
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/models/CompanyUserBillingAddress.java]: Incorrect comment.

File: src/main/java/com/plenigo/sdk/models/ElementList.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package com.plenigo.sdk.models;
+
+import java.util.List;
+
+/**
+ * Created by rtorres on 10/02/16.
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/models/ElementList.java]: Incorrect comments all over the file.

File: src/main/java/com/plenigo/sdk/models/Transaction.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.plenigo.sdk.models;
+
+/**
+ * Created by rtorres on 10/02/16.
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/models/Transaction.java]: Incorrect comments all over the file.

File: src/main/java/com/plenigo/sdk/models/TransactionList.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package com.plenigo.sdk.models;
+
+/**
+ * Created by rtorres on 10/02/16.
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/models/TransactionList.java]: Incorrect comments all over the file.

File: src/main/java/com/plenigo/sdk/services/CompanyService.java [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+     * @throws PlenigoException whenever an error happens
+     */
+    public static ElementList<CompanyUser> getUserList() throws PlenigoException {
+        Map<String, Object> params = new HashMap<String, Object>();
+        Map<String, Object> objectMap = HttpConfig.get().getClient().get(PlenigoManager.get().getUrl(), ApiURLs.COMPANY_USERS, SdkUtils.buildUrlQueryString(params)
+                , JWT.generateJWTTokenHeader(PlenigoManager.get().getCompanyId(), PlenigoManager.get().getSecret()));
+        return new ElementList<CompanyUser>(0, 0, null);
+    }
+
+//    /**
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/services/CompanyService.java]: Why is so many code commented out? Delete it or uncomment it.

File: src/main/java/com/plenigo/sdk/services/TransactionService.java [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+    /**
+     * Default constructor.
+     */
+    private TransactionService() {
+
+    }
+
+
+    public static ElementList<CompanyUser> getUserList(PageRequest request) throws PlenigoException {
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/services/TransactionService.java]: Missing comment.

File: src/main/java/com/plenigo/sdk/services/TransactionService.java [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        ValidationUtils.validate(request);
+        params.put(ApiParams.PAGE_SIZE, request.getPageSize());
+        params.put(ApiParams.PAGE_NUMBER, request.getPageNumber());
+        Map<String, Object> objectMap = HttpConfig.get().getClient().get(PlenigoManager.get().getUrl(), ApiURLs.COMPANY_USERS, buildUrlQueryString(params)
+                , JWT.generateJWTTokenHeader(PlenigoManager.get().getCompanyId(), PlenigoManager.get().getSecret()));
+        return buildElementListForCompanyUsers(objectMap, request.getPageNumber());
+    }
+
+
+    public static List<CompanyUser> getUserList(List<String> userList) throws PlenigoException {
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/services/TransactionService.java]: Missing comment.

File: src/main/java/com/plenigo/sdk/services/TransactionService.java [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+        Map<String, Object> params = new HashMap<String, Object>();
+        params.put(ApiParams.USER_IDS, toCsv(userList));
+        Map<String, Object> objectMap = HttpConfig.get().getClient().get(PlenigoManager.get().getUrl(), ApiURLs.COMPANY_USERS_SELECT, buildUrlQueryString(params)
+                , JWT.generateJWTTokenHeader(PlenigoManager.get().getCompanyId(), PlenigoManager.get().getSecret()));
+        Object companyUsersObj = objectMap.get(ApiResults.ELEMENTS);
+        return buildCompanyUserList(companyUsersObj);
+    }
+
+    private static ElementList<CompanyUser> buildElementListForCompanyUsers(Map<String, Object> objectMap, int pageNumber) {
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/services/TransactionService.java]: Missing comment.

File: src/main/java/com/plenigo/sdk/services/TransactionService.java [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+        int pageSize = 0;
+        if (!pageSizeStr.isEmpty()) {
+            pageSize = Integer.parseInt(pageSizeStr);
+        }
+        Object companyUsersObj = objectMap.get(ApiResults.ELEMENTS);
+        List<CompanyUser> companyUsersList = buildCompanyUserList(companyUsersObj);
+        return new ElementList<CompanyUser>(pageNumber, pageSize, totalElements, companyUsersList);
+    }
+
+    private static List<CompanyUser> buildCompanyUserList(Object companyUsersObj) {
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/services/TransactionService.java]: Missing comment.

File: src/main/java/com/plenigo/sdk/services/TransactionService.java [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+                String agreementState = getValueIfNotNull(companyUser, ApiResults.AGREEMENT_STATE);
+                CompanyUserBillingAddress billingAddress = buildBillingAddressInfo(companyUser);
+                companyUsersList.add(new CompanyUser(customerId, email, userName, language, gender, firstName, name, mobileNumber, userState, birthday, postCode
+                        , street, additionalAddressInfo, city, state, country, agreementState, billingAddress));
+            }
+        }
+        return companyUsersList;
+    }
+
+    private static CompanyUserBillingAddress buildBillingAddressInfo(Map<String, Object> companyUser) {
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/services/TransactionService.java]: Missing comment.

File: src/test/java/com/plenigo/sdk/models/CompanyUserBillingAddressTest.java [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * <p>
+ * Tests for {@link CompanyUser}.
+ * </p>
+ */
+public class CompanyUserBillingAddressTest {
```
Comment (maximilianschweitzer) [inline src/test/java/com/plenigo/sdk/models/CompanyUserBillingAddressTest.java]: Incorrect comments all over the file.

File: src/main/java/com/plenigo/sdk/models/CompanyUser.java (hunk 1 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/models/CompanyUser.java]: fixed

File: src/main/java/com/plenigo/sdk/models/CompanyUserBillingAddress.java (hunk 2 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/models/CompanyUserBillingAddress.java]: fixed

File: src/main/java/com/plenigo/sdk/models/ElementList.java (hunk 3 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/models/ElementList.java]: fixed

File: src/main/java/com/plenigo/sdk/models/Transaction.java (hunk 4 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/models/Transaction.java]: fixed

File: src/main/java/com/plenigo/sdk/models/TransactionList.java (hunk 5 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/models/TransactionList.java]: fixed

File: src/main/java/com/plenigo/sdk/services/CompanyService.java (hunk 6 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/services/CompanyService.java]: fixed

File: src/main/java/com/plenigo/sdk/services/TransactionService.java (hunk 7 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/services/TransactionService.java]: fixed

File: src/main/java/com/plenigo/sdk/services/TransactionService.java (hunk 8 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/services/TransactionService.java]: fixed

File: src/main/java/com/plenigo/sdk/services/TransactionService.java (hunk 9 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/services/TransactionService.java]: fixed

File: src/main/java/com/plenigo/sdk/services/TransactionService.java (hunk 10 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/services/TransactionService.java]: fixed

File: src/main/java/com/plenigo/sdk/services/TransactionService.java (hunk 11 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/services/TransactionService.java]: fixed

**Pull request merged**